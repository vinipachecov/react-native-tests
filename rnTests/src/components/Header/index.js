import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { theme } from '../../theme';

const Header = () => (
  <View style={headerStyles.view}>
    <Text style={headerStyles.text}>My bills</Text>
  </View>
);

const headerStyles = StyleSheet.create({
  view: {
    alignSelf: 'flex-start',
    backgroundColor: theme.colors.primary.default,
    display: 'flex',
    height: 128,
    justifyContent: 'flex-end',
    padding: theme.spaces[2],
    width: '100%',
  },
  text: {
    color: theme.colors.white,
    fontSize: theme.fontSizes.headerOne,
    fontWeight: 'bold',
    marginTop: theme.spaces[1],
  },
});

export default Header;
