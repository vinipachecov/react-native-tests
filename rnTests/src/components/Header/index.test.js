import React from 'react';
import renderer from 'react-test-renderer';

import Header from './index';


// Como é um componente stateless e não recebe nenhuma prop
// será usado o teste com snapshot
test('[Component] Header matches snapshots', () => {
  const tree = renderer.create(<Header/>).toJSON();
  //Lembrando que o snapshot tira uma "foto" da tela e confere como está o código
  expect(tree).toMatchSnapshot();
});