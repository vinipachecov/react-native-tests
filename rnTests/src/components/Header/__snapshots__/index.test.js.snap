// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Header matches snapshots 1`] = `
<View
  style={
    Object {
      "alignSelf": "flex-start",
      "backgroundColor": "#0d47a1",
      "display": "flex",
      "height": 128,
      "justifyContent": "flex-end",
      "padding": 8,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 36,
        "fontWeight": "bold",
        "marginTop": 4,
      }
    }
  >
    My bills
  </Text>
</View>
`;
